src/common/smartPay.ts

```diff
@@ -0,0 +1,475 @@
+      const decode = await cleanMultiFetch(
+        asSmartPayQrDecode,
+        ['https://connect.smartpay.com.vc'],
+        `api/pix/qrdecode?qrcode=${qrcode}`,
+        undefined,
+        undefined,
+        io.fetch
+      )
+      if (decode.status !== 'ok') {
+        throw new Error(decode.msg)
+      }
+      const { data: decodeData } = decode
+      const { amount, key, name, timeout: decodeTimeout } = decodeData
+      if (decodeTimeout > 0 && decodeTimeout < 120) {
+        throw new Error('ErrorPixExpired')
+      }
+
+      let nativeAmount: string | undefined
+      let expireDate: Date | undefined
+      if (amount !== 0) {
+        const paramsObj: QueryParams = {
+          type: 'buy',
+          profile: 'transfer',
+          currency: 'brl',
+          conv: 'txusdt',
+          target: 'amount',
+          user: smartPayUserId,
+          amount: amount === 0 ? 100 : amount
+        }
+        const params = makeQueryParams(paramsObj)
+
+        // Get swap quote
+        const quote = await cleanMultiFetch(
+          asSmartPaySwapQuote,
+          ['https://connect.smartpay.com.vc'],
+          `api/swapix/swapquote?${params}`,
+          undefined,
+          undefined,
+          io.fetch
+        )
+
+        const { data: quoteData } = quote
+        const { amount_txusdt: amountTxusdt, timeout: quoteTimeout } = quoteData
+        if (quoteTimeout > 0 && quoteTimeout < 120) {
+          throw new Error('ErrorPixExpired')
+        }
+
+        // Get USDT info
+        const token = currencyInfo.metaTokens.find(
+          t => t.currencyCode === 'USDT'
+        )
+        if (token == null) return
+        nativeAmount = mul(amountTxusdt, token.denominations[0].multiplier)
+        const timeout = Math.min(
+          decodeTimeout > 0 ? decodeTimeout : MAX_TIMEOUT_S,
+          quoteTimeout > 0 ? quoteTimeout : MAX_TIMEOUT_S
+        )
+        expireDate = new Date(now.getTime() + timeout * 1000)
+      }
+
+      const out: EdgeParsedUri = {
+        currencyCode: 'USDT',
+        metadata: {
+          name,
+          notes: `To PIX: ${key}`
+        },
+        expireDate,
+        nativeAmount,
+        publicAddress: smartPayPublicAddress,
+        uniqueIdentifier: code
+      }
+      return out
+    } catch (e: any) {
+      console.log(`Could not query PIX address ${code}: ${e.message}`)
+    }
+  } else {
+    const [isPix, pixKey] = formatPixKey(code)
+    if (!isPix) return
+    const out: EdgeParsedUri = {
+      currencyCode: 'USDT',
+      metadata: {
+        name: `PIX: ${pixKey}`,
+        notes: `To PIX: ${pixKey}`
+      },
+      publicAddress: 'TUmgPbM5J6om7Z2PJjzrbSEbXit84ZhVCj',
+      uniqueIdentifier: pixKey
+    }
+    return out
+  }
+}
+
+// ---------------------------------------------------------------------------
+// All code below provided by smartpay.com.vc
+// ---------------------------------------------------------------------------
```

Comment:
Can we put all this third-party code into its own file, like `pixKey.ts` or something? That makes it pretty clean in terms of ours vs. theirs.